crates/uv-distribution/src/workspace.rs

```diff
@@ -824,6 +808,54 @@ fn is_excluded_from_workspace(
     Ok(false)
 }
 
+/// An iterator over a [`Pattern`] in a workspace.
+///
+/// Iterates from the workspace root, returning paths that match the pattern.
+#[derive(Debug)]
+struct WorkspaceGlobIter<'a> {
+    root: &'a Path,
+    glob: &'a Pattern,
+    iter: walkdir::IntoIter,
+}
+
+impl<'a> WorkspaceGlobIter<'a> {
+    fn new(root: &'a Path, glob: &'a Pattern) -> Self {
+        Self {
+            root,
+            glob,
+            iter: walkdir::WalkDir::new(root).into_iter(),
+        }
+    }
+}
+
+impl<'a> Iterator for WorkspaceGlobIter<'a> {
+    type Item = Result<PathBuf, walkdir::Error>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            let entry = self.iter.next();
+            let entry = match entry {
+                Some(entry) => entry,
+                None => return None,
+            };
+            let entry = match entry {
+                Ok(entry) => entry,
+                Err(err) => return Some(Err(err)),
+            };
+            if entry.file_type().is_dir() {
+                if let Ok(path) = entry.path().strip_prefix(self.root) {
+                    if self.glob.matches_path(path) {
+                        // Match, but avoid recursing into the directory.
```

Comment:
I have a vague feeling that this isn't correct for some recursive patterns? Like if you had `members = ["crates/**"]`?